// Copyright: 1618labs, Inc. 2011 onwards.
// All rights reserved.

package rpc;
option java_package = "com.whisperlib.test.rpc";
option java_outer_classname = "TextProto";

option cc_generic_services = true;

message TestReq {
    required int32 x = 1;
    required int32 y = 2;
    optional string s = 3;
}

message TestReply {
    required int64 z = 1;
    optional string s = 2;
}

service TestService {
  rpc Mirror(TestReq) returns (TestReq);
  rpc Sum(TestReq) returns (TestReply);
  rpc Multiply(TestReq) returns (TestReply);
}